packages/powersync/bin/setup_web.dart

```diff
@@ -0,0 +1,168 @@
+import 'dart:convert';
+import 'dart:io';
+import 'package:collection/collection.dart';
+import 'package:pub_semver/pub_semver.dart';
+import 'package:pubspec_parse/pubspec_parse.dart';
+import 'package:args/args.dart';
+
+void main(List<String> arguments) async {
+  var parser = ArgParser();
+  // Add a flag to enable/disable the download of worker (defaults to true)
+  // Pass the --no-worker argument to disable the download of the worker
+  // dart run powersync:setup_web --no-worker
+  parser.addFlag('worker', defaultsTo: true);
+  // Add a option to specify the output directory (defaults to web)
+  // Pass the --output-dir argument to specify the output directory
+  // dart run powersync:setup_web --output-dir assets
+  parser.addOption('output-dir', abbr: 'o', defaultsTo: 'web');
+  var results = parser.parse(arguments);
+  bool downloadWorker = results.flag('worker');
+  String outputDir = results.option('output-dir')!;
+
+  final root = Directory.current.uri;
+  print('Project root: ${root.toFilePath()}');
+
+  final wasmPath = '${root.toFilePath()}$outputDir/sqlite3.wasm';
+
+  final workerPath = '${root.toFilePath()}$outputDir/powersync_db.worker.js';
+
+  final packageConfigFile = File.fromUri(
+    root.resolve('.dart_tool/package_config.json'),
+  );
+  dynamic packageConfig;
+  try {
+    packageConfig = json.decode(await packageConfigFile.readAsString());
+  } on FileSystemException {
+    print('Missing .dart_tool/package_config.json');
+    print('Run `flutter pub get` first.');
+    exit(1);
+  } on FormatException {
+    print('Invalid .dart_tool/package_config.json');
+    print('Run `flutter pub get` first.');
+    exit(1);
+  }
+
+  try {
+    final httpClient = HttpClient();
+
+    final powersyncPackageName = 'powersync';
+
+    if (downloadWorker) {
+      final powersyncPkg =
+          getPackageFromConfig(packageConfig, powersyncPackageName);
+
+      final powersyncVersion = getPubspecVersion(
+          packageConfigFile, powersyncPkg, powersyncPackageName);
+
+      final workerUrl =
+          'https://github.com/powersync-ja/powersync.dart/releases/download/powersync-v$powersyncVersion/powersync_db.worker.js';
+
+      await downloadFile(httpClient, workerUrl, workerPath);
+    }
+
+    final sqlitePackageName = 'sqlite3';
+
+    final sqlite3Pkg = getPackageFromConfig(packageConfig, sqlitePackageName);
+
+    String sqlite3Version =
+        "v${getPubspecVersion(packageConfigFile, sqlite3Pkg, sqlitePackageName)}";
+
+    List<String> tags = await getLatestTagsFromRelease(httpClient);
+    String? matchTag = tags.firstWhereOrNull((element) =>
+        element.contains(sqlite3Version) && coreVersionIsInRange(element));
+    if (matchTag != null) {
+      sqlite3Version = matchTag;
+    } else {
+      throw Exception(
+          """No compatible powersync core version found for sqlite3 version $sqlite3Version
```

Comment:
Could we maybe add a hint containing some of the latest supported `sqlite3` versions?